test/snippet/validators_input_file_ext_from_file.cpp

```diff
@@ -12,8 +12,8 @@ int main()
     sharg::input_file_validator validator2{std::vector{std::string{"exe"}, std::string{"fasta"}}};
     std::cerr << validator2.get_help_page_message() << '\n';
 
-    // Give the seqan3 file type as a template argument to get all valid extensions for this file.
-    sharg::input_file_validator<seqan3::sequence_file_input<>> validator3{};
+    // Give the sharg file type as a template argument to get all valid extensions for this file.
+    sharg::input_file_validator validator3{seqan3::sequence_file_input<>::valid_formats};
```

Comment:
I missed testing the snippets and unfortunately I get an error for the new notation:
```bash
Consolidate compiler generated dependencies of target validators_3_snippet
[ 40%] Building CXX object CMakeFiles/validators_3_snippet.dir/validators_3.cpp.o
/Users/lydia/Repos/sharg-parser/test/snippet/validators_input_file_ext_from_file.cpp: In function 'int main()':
/Users/lydia/Repos/sharg-parser/test/snippet/validators_input_file_ext_from_file.cpp:16:88: error: expected primary-expression before '}' token
   16 |     sharg::input_file_validator validator3{seqan3::sequence_file_input<>::valid_formats};
      |                                                                                        ^
/Users/lydia/Repos/sharg-parser/test/snippet/validators_input_file_ext_from_file.cpp:16:88: error: no matching function for call to 'sharg::input_file_validator::input_file_validator(<brace-enclosed initializer list>)'
```

Reply:
ops. I was to quick. It seems like you access the extensions like this:
`seqan3::detail::valid_file_extensions<typename seqan3::sequence_file_input<>::valid_formats>()`

I did not know that there is no nice way yet to access the extensions. I don't remember why the file class itself does not have a member for this. @eseiler Do you remember? There is nothing in the core meeting notes. I feel like someone had a say in this but I just don't why the file shouldn't have a member with the allowed extensions like 
`seqan3::sequence_file_input<>::file_extensions`

As an alternative we could add a helper function
`seqan3::get_file_extensions<seqan3::sequence_file_input<>>()` that just calls the detail function, or make the detail function public.

Reply:
The problem with the helper is that it is `detail`.
I'm not sure why it isn't a member, maybe we never discussed it.

I only remember that we discussed compression extensions.
In the end, I implemented compression extension like this https://github.com/seqan/raptor/blob/45de7f8d3ce5f21a0f5bd9ceb35173acbfa5867d/include/raptor/argument_parsing/validators.hpp#L159-L184

Reply:
I would remove the IO part in this PR from the argument parser and make this a seqan3 issue about how to access the extensions.

Reply:
I created an issue for this:
https://github.com/seqan/seqan3/issues/2927
Feel free to add anything or rewrite it.